hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -308,12 +308,15 @@ public TransactionBody.Builder createNonfungibleApproval(
             @Nullable final EntityId operatorId) {
         final var builder = CryptoApproveAllowanceTransactionBody.newBuilder();
         if (approveWrapper.isFungible()) {
-            builder.addTokenAllowances(TokenAllowance.newBuilder()
+            var tokenAllowance = TokenAllowance.newBuilder()
                     .setTokenId(approveWrapper.tokenId())
                     .setOwner(Objects.requireNonNull(ownerId).toGrpcAccountId())
                     .setSpender(approveWrapper.spender())
-                    .setAmount(approveWrapper.amount().longValueExact())
-                    .build());
+                    .setAmount(approveWrapper.amount().longValueExact());
+            if (ownerId != null) {
+                tokenAllowance.setOwner(ownerId.toGrpcAccountId());
+            }
```

Comment:
We seem to be setting it twice on line 313 and 317

Reply:
fixed